Sources/MapboxNavigation/LaneView.swift

```diff
@@ -98,127 +236,49 @@ open class LaneView: UIView {
-                if indications == .slightLeft {
-                    LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                } else {
-                    LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                }
-            } else if indications.isSuperset(of: [.straightAhead]) {
-                LanesStyleKit.drawLaneStraight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-            } else if indications.isSuperset(of: [.uTurn]) {
-                let flip = !(drivingSide == .left)
-                LanesStyleKit.drawLaneUturn(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: flip)
-            } else if indications.isEmpty && isValid {
-                // If the lane indication is `none` and the maneuver modifier has a turn in it,
-                // show the turn in the lane image.
-                if maneuverDirection == .sharpRight || maneuverDirection == .right || maneuverDirection == .slightRight {
-                    if maneuverDirection == .slightRight {
-                        LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                    } else {
-                        LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                    }
-                } else if maneuverDirection == .sharpLeft || maneuverDirection == .left || maneuverDirection == .slightLeft {
-                    if maneuverDirection == .slightLeft {
-                        LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                    } else {
-                        LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                    }
-                } else {
-                    LanesStyleKit.drawLaneStraight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                }
+        let rankedIndications = indications.ranked(favoring: maneuverDirection)
+        guard let laneConfiguration = LaneConfiguration(rankedIndications: rankedIndications, drivingSide: drivingSide) else {
+            return
+        }
+        
+        switch laneConfiguration {
+        case .straightOnly:
+            LanesStyleKit.drawLaneStraight(frame: bounds, resizing: resizing, primaryColor: appropriateColor)
+        case .slightTurnOnly(side: let side):
+            LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriateColor,
+                                              flipHorizontally: side == .left)
```

Comment:
With the lane configurations in hand, we call into the LanesStyleKit methods, passing in a color based on lane validity (explicitly specified by the route data, not guessed by the logic above) and flipping the paths based on the configuration.